packages/opencode/src/app/app.ts

```diff
@@ -134,3 +134,9 @@ export namespace App {
     )
   }
 }
+
+function projectDirectoryName(path: string): string {
+  // Prefix prevents collision with "global" and with magic file names like "NIL" on Windows.
+  // TODO: Hash the path to prevent collisions.
+  return "_" + path.replace(/[^A-Za-z0-9_]/g, "-")
```

Comment:
can you explain a bit more here the issue on windows? we don't want to add a `_` to all platforms

there's also a separate bug where it shouldn't start with `-`

Reply:
we fixed that seperate bug so there's a conflict now. but basically the folders will look like `home-thdxr-projects-foo`

we also don't want to do a hash so they're somewhat human readable - lmk

Reply:
> can you explain a bit more here the issue on windows? we don't want to add a `_` to all platforms

`:` is not valid in path names, but `path` is `"C:\foo\bar\baz"`. Instead of playing whackamole with bad characters (`\`, `/`, `:`), I decided to implement a whitelist.

> we also don't want to do a hash so they're somewhat human readable - lmk

I wasn't thinking of replacing the whole path with a hash, but rather appending a hash to disambiguate. But it's just a comment for a future feature; I can drop it.